### PR TITLE
Adornment manager and related file changes

### PR DESCRIPTION
**Pull request by akhera99**
Hello!

These are all the changes that were made related to the adornment manager and are completely separate from inline diagnostics.
Pulling into a separate PR to remove it from the inline diagnostics PR.

### CODE REVIEW

File: src/EditorFeatures/Core.Wpf/LineSeparators/LineSeparatorAdornmentManager.cs [hunk 1]
```diff
@@ -9,54 +9,21 @@
...
-        /// It happens when another region of the view becomes visible or there is a change in tags.
-        /// For us the end result is the same - get tags from tagger and update visuals correspondingly.
-        /// </summary>        
-        protected override void UpdateSpans_CallOnlyOnUIThread(NormalizedSnapshotSpanCollection changedSpanCollection, bool removeOldTags)
+        protected override void AddAdornmentsToAdornmentLayer_CallOnlyOnUIThread(NormalizedSnapshotSpanCollection changedSpanCollection)
         {
-            Contract.ThrowIfNull(changedSpanCollection);
-
-            // this method should only run on UI thread as we do WPF here.
-            Contract.ThrowIfFalse(TextView.VisualElement.Dispatcher.CheckAccess());
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core.Wpf/LineSeparators/LineSeparatorAdornmentManager.cs:37]: i would leave this call in here.

File: src/EditorFeatures/Core.Wpf/LineSeparators/LineSeparatorAdornmentManager.cs [hunk 2]
```diff
@@ -69,40 +36,19 @@ protected override void UpdateSpans_CallOnlyOnUIThread(NormalizedSnapshotSpanCol
...
-                        continue;
-                    }
-
-                    if (!viewLines.IntersectsBufferSpan(span))
-                    {
-                        // span is outside of the view so we will not get geometry for it, but may 
-                        // spent a lot of time trying.
-                        continue;
-                    }
+                    TryMapToSingleSnapshotSpan(tagMappingSpan.Span, TextView.TextSnapshot, out var span);
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core.Wpf/LineSeparators/LineSeparatorAdornmentManager.cs:44]: you aren't cehcking the result of this.

Comment (akhera99) [reply src/EditorFeatures/Core.Wpf/LineSeparators/LineSeparatorAdornmentManager.cs]: Added

File: src/EditorFeatures/Core.Wpf/LineSeparators/LineSeparatorAdornmentManager.cs (hunk 1 above)
Comment (akhera99) [reply src/EditorFeatures/Core.Wpf/LineSeparators/LineSeparatorAdornmentManager.cs:37]: Added back

File: src/EditorFeatures/Core.Wpf/LineSeparators/LineSeparatorAdornmentManager.cs [hunk 3]
```diff
@@ -13,50 +13,21 @@
...
-            Contract.ThrowIfNull(changedSpanCollection);
-
             // this method should only run on UI thread as we do WPF here.
             Contract.ThrowIfFalse(TextView.VisualElement.Dispatcher.CheckAccess());
 
             var viewSnapshot = TextView.TextSnapshot;
-            var visualSnapshot = TextView.VisualSnapshot;
-
             var viewLines = TextView.TextViewLines;
-            if (viewLines == null || viewLines.Count == 0)
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core.Wpf/LineSeparators/LineSeparatorAdornmentManager.cs:43]: Shouldn't we still have these checks?  Or is that already handled elsewhere?

**Review by CyrusNajmabadi — APPROVED** [review]
Lgtm as long as we're still checking for those cases somewhere else.

File: src/EditorFeatures/Core.Wpf/LineSeparators/LineSeparatorAdornmentManager.cs (hunk 3 above)
Comment (akhera99) [reply src/EditorFeatures/Core.Wpf/LineSeparators/LineSeparatorAdornmentManager.cs:43]: Handled at the AbstractAdornmentManager

**Pull request merged**